### ISSUE TITLE
508-defect-1 ❗ Preparer instructions isn't announced to screen reader users; page needs discovery and a potential redesign

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#48302]
## Point of contact

Josh Kim

## Severity level

1, Launchblocker. Critical. Must be fixed before launch.

## Details

Screen reader users in forms mode (e.g. using the tab key as they anticipate only form fields, not static text fields) will likely miss the note describing preparer instructions upon selecting "someone else" during the contact information steps.

<img width="725" alt="Screen Shot 2022-10-17 at 12 29 16 PM" src="https://user-images.githubusercontent.com/14154792/196232268-7f0bd017-67dd-4056-89a0-1c970127c04e.png">

Other issues include:
- Content needs review as the original question does not hint to or directly connect "someone else" to a "preparer"
- Unlike the applicant and service member information, _all_ the preparer's information is presented all at once together within a conditional and expandable field
- If a user changes their mind on the options, they may lose previously filled in information

https://user-images.githubusercontent.com/14154792/196237955-b601b004-96ef-49c8-beae-06076d48bc4d.mov

## Reproduction steps

1. [Begin the pre need flow unsigned in](https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/applicant-information)
2. Select "other" for relationship to service member
3. Reach step 6 of 7: contact information

### Plain language and one thing per page

Instead of conditionally revealing instructions (which won't be picked up by a screen reader) and an entire set of new questions on the same page, I strongly recommend finding a way to reduce its complexity by splitting it apart into separate, more manageable pages.

Content should be reviewed for consistency. I'd recommend talking to the content team and the 10-10ez team (who I believe did something similar to preparers in the past).

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by ccruzgra** [comment]
@joshkimux These background-color only blue info alerts are commonly used in forms throughout VA.gov to reveal information after a user makes a selection. Is the screen reader skipping the info alert instructions entirely after "Someone else" is selected? 
See another example in the VET TEC application where the alert is shown after the user selects "No":
<img width="671" alt="Screenshot 2023-02-14 at 1 21 49 PM" src="https://user-images.githubusercontent.com/41477797/218824390-09960ac8-5d42-4ff2-ac82-0b9df47a8527.png">

**Comment by joshkimux** [comment]
> These [background-color-only blue info alerts](https://design.va.gov/components/alert#background-color-only-alert) are commonly used in forms throughout VA.gov to reveal information after a user makes a selection. Is the screen reader skipping the info alert instructions entirely after "Someone else" is selected?

This is a great question @ccruzgra , it depends! 

**If no `aria-live` is used**, _all_ the information will almost certainly be missed by screen reader users. When screen reader users navigate form fields, they usually use forms mode, which means they will only have access to interactive components. When non-interactive information (like added content in a `p` element) is sandwiched between interactive components, it will be skipped completely unless the user has the clairvoyance to navigate outside of forms mode backwards to catch it.

**If `aria-live` is used for this blue info alert**, it still may not be sufficient as:
- It's a relatively long information alert with multiple bullets 
- It's not clear _where_ that alert is being read from

As a result, a screen reader user in forms mode may hear it but:
- forget the details and
- not know how to access the information again (unless they navigate the page line by line from the top since it's unclear where the new information has been added-- this can easily take 5-10m to do )

[This pattern of conditionally revealing static information is one of the key reasons why the wizard got deprecated](https://design.va.gov/patterns/wizards#retrofit-this-pattern). VET TEC may not have been reviewed since this new guidance came into place, as we typically recommend to add additional information using the following methods:
- Ideal state: divide the conditionals into separate pages. For example, ask who is filling out the application, then on the _next_ page provide more details and fields based on the selection.
- Band aid: provide an additional information component to provide additional details as it is interactive and will take focus (meaning it effectively won't be missed by screen readers).

For more support, I strongly recommend reaching out to @coforma-terry and the CAIA team ❤️

**Comment by ccruzgra** [comment]
@joshkimux Thanks for the follow up! That was my next question regarding if we could the additional info as a temporary fix until we can do proper discovery on this page. Glad that it's an option.

**Comment by ccruzgra** [comment]
@joshkimux Just to confirm, for the bandaid,  we should be using an additional info component like in this example? Thank you.

<img width="749" alt="Screenshot 2023-02-14 at 2 51 43 PM" src="https://user-images.githubusercontent.com/41477797/218846270-5075a00d-431f-461b-90da-5dca5f05ccfe.png">

**Comment by joshkimux** [comment]
@ccruzgra , [you can use the additional info component on the design system here](https://design.va.gov/components/additional-info) provided there's documentation on this being a temporary fix until a more systemic solution can be arranged ✨

**Comment by coforma-terry** [comment]
Hey All - sure, absolutely happy to help with this - best thing to do would be to fill out an [intake request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=RLHecht%2C+coforma-terry&labels=sitewide+CAIA%2C+sitewide+content-product+support%2C+Sitewide+IA%2C+sitewide+content&template=sitewide-content-intake-form.md&title=%3CType+of+Request%3E+from+%3CTeam%3E) for us so it's officially on our radar :)

**Comment by AlexanderSherrow** [comment]
@joshkimux @coforma-terry did that intake request ever get submitted? Currently working on this issue, was going to proceed with the additional info component bandaid solution if no other decisions were made.

**Comment by coforma-terry** [comment]
Hi @AlexanderSherrow - it doesn't look like we got an intake request for this ^^

**Comment by AlexanderSherrow** [comment]
@joshkimux im applying the additional-info component solution now. Do we have any idea what we want the trigger text to be?

**Comment by joshkimux** [comment]
@AlexanderSherrow Hi Alex, that's a great question! Do you have a designer or content writer on your team that can support? If not, I recommend reaching out to:
- [platform content and design office hours](https://depo-platform-documentation.scrollhelp.site/support/office-hours)
- OR, filing an intake request with Terry for the CAIA team (shared support for content, IA, and accessibility)

**Comment by coforma-terry** [comment]
Intake request received, thanks!

**Comment by sonntag-adam** [comment]
@joshkimux fix is in staging

**Comment by joshkimux** [comment]
@SarahKay8 if you have time, could you validate this 🙏

**Comment by SarahKay8** [comment]
Validated additional info component has successfully been added to form. Closing.
Tools - VO + Chrome + Axe . VO + Safari.
